subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/DefaultTaskExecutionModeResolver.java

```diff
@@ -55,4 +62,13 @@ public TaskExecutionMode getExecutionMode(TaskInternal task, TaskProperties prop
 
         return TaskExecutionMode.INCREMENTAL;
     }
+
+    private static boolean requiresInputChanges(TaskInternal task) {
+        for (InputChangesAwareTaskAction taskAction : task.getTaskActions()) {
+            if (taskAction instanceof AbstractIncrementalTaskAction) {
```

Comment:
I wonder if we should break this already for `InputChanges` and only add the deprecation for `IncrementalTaskInputs`. But given that we'll only have this for one minor version before we remove it for both it's probably not worth the trouble.

Reply:
That is what I thought as well.